roslibrust/src/rosbridge/publisher.rs

```diff
@@ -12,6 +12,7 @@ use crate::{Client, RosLibRustResult, RosMessageType};
 ///   - Ability for publish to by const
 // Instead of giving back a publisher should we give back a reference to one, and give back the
 // same reference when you advertise multiple times? Would require non-mut references?
+#[derive(Clone)]
```

Comment:
Uhh... I think we can't do this...

This will break unadvertise logic? If we want publisher to be clone we need to keep track of the count of outstanding publishers for a topic and only unadvertise the topic when the count drops to zero. There is a #20 issue to do this.

Reply:
Ah good call, will remove.